### ISSUE TITLE
BUG-000109471 Attachments added in Crowdsource App disappear after saving the edits.

### ISSUE THREAD

**Issue by allisonmuise** [Esri/crowdsource-manager#247]
<div><div><ol><li>Add data to ArcGIS Pro.</li><li>Publish the data to ArcGIS Online.</li><li>Login to ArcGIS Online in web browser. </li><li>Open the feature service and save it as a web map. </li><li>Click on Group tab. </li><li>Create a group. </li><li>Now share the service definition, hosted feature service and web map in the group.</li><li>Create a web app from the group (Overview &gt; Share &gt; Create A Web App &gt; Crowdsource Manager). </li><li>Configure the app. Make sure the box &#39;Display related records using the popup configuration&#39; in the &#39;Reports&#39; tab is checked. </li><li>Now select a feature and add a related record. Add an attachment to this related record. </li><li>Click save and view your result. The attachment you just added is missing.</li></ol>
Below are the links for hosted feature service, web map and web application on sample data. <br><br><b>Hosted Feature Service:</b><br><a href="http://ess.maps.arcgis.com/home/item.html?id=1c9d809154054331b8c6cebe2183858b" target="_blank">http://ess.maps.arcgis.com/home/item.html?id=1c9d809154054331b8c6cebe2183858b</a><br><br><b>Web Map:</b><br><a href="http://ess.maps.arcgis.com/home/item.html?id=d17e09827a4f475fa4f8b499cfd054aa" target="_blank">http://ess.maps.arcgis.com/home/item.html?id=d17e09827a4f475fa4f8b499cfd054aa</a><br><br><b>Web Application:</b><br><a href="http://ess.maps.arcgis.com/home/item.html?id=7b988765921a483aa0ca79af133a77aa" target="_blank">http://ess.maps.arcgis.com/home/item.html?id=7b988765921a483aa0ca79af133a77aa</a></div><div>
<span><b>Salesforce ID:</b> BUG-000109471</span><br/>
<span><b>Salesforce Submitter:</b> Janhavi Hindlekar</span><br/>
<span><b>Salesforce Submit Date:</b> 11/13/2017 2:09 AM</span><br/>
<span><b>Salesforce Bug Type:</b> Failure/Error</span><br/>
<span><b>Salesforce Severity:</b> Medium</span><br/>
<span><b>Repro Data:</b> (n/a)</span><br/>
<span><b>Work Around:</b> (n/a)</span><br/>
<span><b>Product:</b> ArcGIS Online</span><br/>
<span><b>Functional Category:</b> ArcGIS Online Application Templates</span><br/>
<span><b>Client Platform:</b> (n/a)</span><br/>
<span><b>Version Found:</b> N/A</span><br/>
<span><b>Planned Version Fixed:</b> (n/a)</span><br/>
<span><b>Comment:</b> (n/a)</span><br/>
</div></div>

**Comment by allisonmuise** [comment]
#1914

**Comment by allisonmuise** [comment]
@CTLocalGovTeam I have re-shared the app, map, and layer listed above so you should now be able to reproduce the issue using the provided steps. I still see the issue, although I am not able to reproduce it on all applications.

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

After analysing, we noticed that this issue is related to table itself as described below.

We opened the webmap in map viewer and added the attachments to the table. It gets uploaded successfully, however doesn't displays in view tab. In this state, once we refresh the webmap, attachment gets removed. 

![csm_edit_attachment_1](https://user-images.githubusercontent.com/2959186/34720236-fb2faefc-f563-11e7-8b34-5f52f767b213.gif)

Hence, when attachment query is executed it returns 0 results.

![image](https://user-images.githubusercontent.com/2959186/34720368-72f418ba-f564-11e7-99a5-9470925b9743.png)

We tried to use [another layer](https://services.arcgis.com/Wl7Y1m92PbjtJs5n/ArcGIS/rest/services/RelatedTabel_HydrantInspections/FeatureServer/1) from the same service directory, and it's working correctly in map viewer and application.

![csm_edit_attachment_3](https://user-images.githubusercontent.com/2959186/34721596-5073b8a4-f569-11e7-93d9-88b528da55f0.gif)


So, this seems to be an issue with the layer itself and we don't have an control from code or API



--Pratik Gohil
